hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/systemcontracts/hts/update/UpdateTranslator.java

```diff
@@ -86,8 +87,13 @@ public boolean matches(@NonNull HtsCallAttempt attempt) {
 
     @Override
     public Call callFrom(@NonNull HtsCallAttempt attempt) {
-        return new DispatchForResponseCodeHtsCall(
-                attempt, nominalBodyFor(attempt), UpdateTranslator::gasRequirement, UpdateDecoder.FAILURE_CUSTOMIZER);
+        final var transactionBody = nominalBodyFor(attempt);
+        return new SchedulableDispatchForResponseCodeHtsCall(
```

Comment:
If we do this, will this not affect the way `updateTokenInfo` works as we probably do not want a `SchedulableDispatchForResponseCodeHtsCall` in that case?

Reply:
`SchedulableDispatchForResponseCodeHtsCall` is only providing an implementation for `asSchedulableDispatchIn` so it shouldn't add any other change to the current behaviour of the update

Reply:
I see.   

Consider whether it would be  better to not introduce the `SchedulableDispatchForResponseCodeHtsCall` class but rather implement yet another constructor to `DispatchForResponseCodeHtsCall` which takes a @Nullable `SchedulableTransactionBody` that could be returned by the `asSchedulableDispatchIn` method.  The method could check for null and throw the same error as in the `Call` interface if the ``SchedulableTransactionBody` is null.

I'm could be convinced that this not necessarily better but I think I would be less confused than the introduction of this new class (and when to use which class).